core/src/replay_stage.rs

```diff
@@ -832,6 +863,83 @@ impl ReplayStage {
         }
     }
 
+    fn retransmit_not_propagated(
+        poh_recorder: &Arc<Mutex<PohRecorder>>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+        retransmit_slots_sender: &RetransmitSlotsSender,
+        progress: &mut ProgressMap,
+        last_retransmit_retry_info: &mut LastRetransmitInfo,
+    ) -> Option<Slot> {
+        let mut retransmitted_slot = None;
+        let start_slot = poh_recorder.lock().unwrap().start_slot();
+
+        if let Some(latest_leader_slot) = progress.get_latest_leader_slot(start_slot) {
+            let bank = bank_forks
+                .read()
+                .unwrap()
+                .get(latest_leader_slot)
+                .expect(
+                    "In order for propagated check to fail, \
+                        latest leader must exist in progress map, and thus also in BankForks",
+                )
+                .clone();
+            if last_retransmit_retry_info.slot == bank.slot() {
```

Comment:
maybe a unit test to show:
1. On different slots the `last_retransmit_retry_info` is being reset
2. Manipulate combinations of elapsed time + retry_iteration in `last_retransmit_retry_info` to show the backoff is working as expected.

Reply:
added unit test